### PR TITLE
implement Date interface in dynamic-simple

### PR DESCRIPTION
**Pull request by brucehuke**

### CODE REVIEW

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c [hunk 1]
```diff
@@ -6,39 +6,137 @@
...
+  if (second < 0 || second > 59) {
+    return -1;
+  }
+  tm_.tm_sec = second;
+
+  tm_.tm_year = year - 1900;
+  tm_.tm_mon = month - 1;
+  tm_.tm_mday = day ;
+  tm_.tm_isdst = 0;
+  *t = mktime(&tm_);
```
Comment (xujuntwt95329) [inline runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c:43]: Should check return value of `mktime`

```
On error, mktime() returns the value (time_t) -1.
```

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/dyn_class.c [hunk 2]
```diff
@@ -98,7 +98,7 @@ find_class_constructor(const char *name)
 
     for (i = 0; i < DynClassEnd; i++) {
         meta = class_meta_array[i];
-        if (!meta)
+        if (!meta || !meta->name)
```
Comment (xujuntwt95329) [inline runtime-library/libdyntype/dynamic-simple/dyn-value/class/dyn_class.c:101]: If the name field is necessary, we should `assert(meta->name != NULL)` at before L104, and we should add name for all class_metas

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c [hunk 3]
```diff
@@ -6,39 +6,137 @@
...
+ * @brief char* to time_t
+ *
+ * @param str year-month-day hour:minute:second, like
+ * "2023-12-23 15:58:12"
+ * @param t time_t addr
+ * @return int 0: success
+ */
+int strtotime(const char *str, time_t *t) {
+  int year, month, day, hour, minute, second;
+  struct tm tm_;
```
Comment (xujuntwt95329) [inline runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c:19]: Had better use clang-format to format the code

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c (hunk 1 above)
Comment (brucehuke) [reply runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c]: Done

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/dyn_class.c (hunk 2 above)
Comment (brucehuke) [reply runtime-library/libdyntype/dynamic-simple/dyn-value/class/dyn_class.c]: Done

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c (hunk 3 above)
Comment (brucehuke) [reply runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c]: Done

File: runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c [hunk 4]
```diff
@@ -6,39 +6,141 @@
...
 
-    if (!init_dyn_object(dyn_obj, DynClassDate)) {
+    if (!init_dyn_object((DyntypeObject*)dyn_obj, DynClassDate)) {
         wasm_runtime_free(dyn_obj);
         return NULL;
     }
 
+    if(argc==0){
+        dyn_obj->time = time(NULL);
+    }else if (argc == 1 && argv[0]->class_id == DynClassString){
```
Comment (xujuntwt95329) [inline runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c:68]: These lines seems not formatted

Comment (brucehuke) [reply runtime-library/libdyntype/dynamic-simple/dyn-value/class/date.c]: Done

**Review by xujuntwt95329 — APPROVED** [review]
LGTM

**Pull request merged**